cli/crates/typed-resolvers/src/error.rs

```diff
@@ -3,13 +3,17 @@ use std::fmt;
 
 #[derive(Debug)]
 pub enum CodegenError {
```

Comment:
This isn't necessarily specific to this PR, but why not use this error for these?

Reply:
To avoid dependencies where they're not strictly necessary, but I can definitely change to thiserror for uniformity in this PR or later if that sounds better.

Reply:
Well the CLI already has thiserror so it should theoretically be fine

Reply:
Whatever you prefer in terms of where to do it

Reply:
done and pushed